### DISCUSSION TITLE
Word wrap configuration?

### DISCUSSION THREAD

**Discussion by VANADAIN** [helix-editor/helix#2108] [Q&A] [answered]
Hi,
I installed fresh version of helix recently and can't find info how to set word wrap of text-width functionality (like limitation in N columns for editor space). Is this configuration option possible?

**Comment by the-mikedavis** [comment, marked as answer]
This is not yet implemented, see https://github.com/helix-editor/helix/issues/136

**Reply by giuliano-macedo** [reply]
This answer is outdated, can we change to https://github.com/helix-editor/helix/discussions/2108#discussioncomment-8812144?

**Reply by lmmx** [reply]
Indeed, I've just been using this happily, can confirm :-) But specifically I'd suggest [sicher's additional settings to turn the wrap indicator off](https://github.com/helix-editor/helix/discussions/2108#discussioncomment-14089928) etc. (otherwise you get a disruptive wrapping arrow at the end of line I think)

**Comment by lynnpepin** [comment]
My understanding is that [this has now been merged](https://github.com/helix-editor/helix/pull/5420), but it's unclear to me how to activate it. Searching for it simply brings up this discussion and other issues.

I ended up finding it here: https://helix-editor.com/news/release-23-03-highlights/ so I wanted to add it to this discussion

```toml
# ~/.config/helix/config.toml
[editor.soft-wrap]
enable = true
```

**Reply by zenlor** [reply]
Another little note, to enable it at runtime:

```
:set-option soft-wrap.enable true
```

**Comment by peterjc** [comment]
There isn't a way to set this by language, is there?

e.g. Line wrapping on in Markdown and LaTeX, but off by default.

Or, line wrapping off in Python and R, but on by default.

**Reply by the-mikedavis** [reply]
You can override the editor-level `soft-wrap` config in `languages.toml` for a language: https://docs.helix-editor.com/master/languages.html#language-configuration

**Reply by lmmx** [reply]
That didn't work for me with a `[[language]]` and `name = "markdown"` for some reason (but no matter, I'm fine setting it globally!)

**Reply by sicher** [reply]
I have this in my languages.toml:

```
[[language]]
name = "markdown"
text-width = 80
soft-wrap = { enable = true, wrap-at-text-width = true, max-wrap = 25, wrap-indicator = "" }

```

**Reply by lmmx** [reply]
oh that is much nicer, you don't get the little down-right hook arrows with the `""` wrap-indicator, smart. Thanks for sharing!
